src/IotHub/IotHub/IotHub/NewAzureRmIotHubKey.cs

```diff
@@ -0,0 +1,123 @@
+﻿// ----------------------------------------------------------------------------------
+//
+// Copyright Microsoft Corporation
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+namespace Microsoft.Azure.Commands.Management.IotHub
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Globalization;
+    using System.Linq;
+    using System.Management.Automation;
+    using System.Text;
+    using Microsoft.Azure.Commands.Management.IotHub.Common;
+    using Microsoft.Azure.Commands.Management.IotHub.Models;
+    using Microsoft.Azure.Management.IotHub;
+    using Microsoft.Azure.Management.IotHub.Models;
+    using ResourceManager.Common.ArgumentCompleters;
+
+    [Cmdlet("New", ResourceManager.Common.AzureRMConstants.AzureRMPrefix + "IotHubKey", SupportsShouldProcess = true)]
```

Comment:
@anusapan I think we should add two additional parameter sets to this cmdlet to allow users to provide the identity values of the key in different forms:

(1) A parameter set with an `-InputObject` parameter that's the same type as the returned object from `Get-AzIotHubKey`. This will allow users to get the key that they want to generate a new value for using `Get-AzIotHubKey` and then pipe it to this cmdlet without needing to provide a value for `-ResourceGroupName`, `-Name`, and `-KeyName`

(2) A parameter set with a `-ResourceId` parameter of type `string`. This will allow users to get a generic resource object from `Get-AzResource` and pipe it to this cmdlet without needing to provide a value for the same three parameters.

Let me know if you have any questions about these scenarios. I can provide example implementations of either of these.

Reply:
Added ResourceId parameter set and for InputObject parameter set the expected values aren't exposed by the related cmdlet